Quick-Start/Anatomy_of_extension.md

```diff
@@ -1,418 +1,418 @@
-#### The anatomy of a PostgreSQL extension consists of various components that work together to define and extend the functionality of the database. Here are the key elements:
+## The anatomy of a PostgreSQL extension consists of various components that work together to define and extend the functionality of the database. Here are the key elements: 
+ 
+# 1) SQL-Script File:  
+A SQL script file is a text file that contains a series of SQL statements used to define the objects and behavior of a PostgreSQL extension. These statements can include creating tables, defining functions, creating data types, setting up permissions, and performing various database operations. 
+ 
+The role of a SQL script file in PostgreSQL extensions is crucial. It serves the following purposes: 
+ 
+1.  **Defining extension objects**: A SQL script file is used to define the necessary database objects specific to the extension, such as tables, views, functions, triggers, and operators. These objects define the functionality and behavior of the extension within the PostgreSQL database. 
+ 
+2.  **Initializing extension data**: The SQL script file can also include SQL statements to populate initial data into the extension's tables or perform any required data setup during the extension installation process. 
+ 
+3.  **Providing extension functionality**: The SQL script file contains SQL statements that implement the desired functionality of the extension. For example, it can include function definitions that perform complex calculations, data manipulation, or other operations specific to the extension's purpose. 
+ 
+4.  **Managing dependencies**: When creating a PostgreSQL extension, there might be dependencies on other extensions or objects. The SQL script file can include statements to ensure that these dependencies are properly managed, such as checking for the existence of required extensions or objects before creating or modifying them. 
+ 
+5.  **Version management**: SQL script files are often used to manage versioning and updates of the extension. By including version-specific SQL statements in the script file, developers can handle schema modifications or data migrations required when upgrading the extension to a new version. 
+ 
+6.  **Reproducibility and portability**: SQL script files provide a standardized and portable way to define the extension's structure and behaviour. They can be easily shared with other developers or users, allowing them to reproduce the extension in their PostgreSQL environments. 
+ 
+7.  **Documentation and reference**: SQL script files serve as a valuable source of documentation and reference for the extension. Developers can review the script file to understand the extension's structure, dependencies, and functionality. It helps in troubleshooting, maintaining, and extending the extension in the future. 
```

Comment:
Reference yes, but documentation less so. One usually includes docs in a separate file, unless one is using embedded documentation, as with [pg_readme](https://pgxn.org/dist/pg_readme/).